front/pages/w/[wId]/assistant/new.tsx

```diff
@@ -222,235 +232,170 @@ export default function AssistantNew({
     setShowQuickGuide(false);
   });
 
+  const scrollToInputBar = useCallback(() => {
+    setTimeout(() => {
+      const scrollContainerElement = document.getElementById(
+        "assistant-input-header"
+      );
+      if (scrollContainerElement) {
+        scrollContainerElement.scrollIntoView({ behavior: "smooth" });
+      }
+    }, 50); // Allows browser to complete the layout update before scrolling.
+  }, []);
+
+  const handleAssistantClick = useCallback(
+    (agent: LightAgentConfigurationType) => {
+      scrollToInputBar();
+      setSelectedAssistant({
+        configurationId: agent.sId,
+      });
+
+      setTimeout(() => {
```

Comment:
The `setTimeout` that impacts the UI should be moved to a `useEffect`.

Reply:
I reckon I did not check thoroughly those 2 either; so I'll give that a deeper look (from afar it seemed ok)